File: Python/516. Longest Palindromic Subsequence.py
class Solution:
    def longestPalindromeSubseq(self, s):
        """
        :type s: str
        :rtype: int
        """
        dp = [[0 for i in range(len(s))] for j in range(len(s))]
        for j in range(len(s)):
            dp[j][j] = 1
            for i in range(j-1, -1, -1):
                if s[i] == s[j]:
                    dp[i][j] = dp[i+1][j-1] + 2
                else:
                    dp[i][j] = max(dp[i+1][j], dp[i][j-1])
        return dp[0][len(s)-1]

File: C++/516. Longest Palindromic Subsequence.cpp
class Solution {
public:
    int longestPalindromeSubseq(string s) {
        vector<vector<int> >dp(s.length(),vector<int>(s.length()));
        for(int j=0;j<s.length();j++)
        {
            dp[j][j]=1;
            for(int i=j-1;i>=0;i--)
            {
                if(s[i]==s[j])
                    dp[i][j] = dp[i+1][j-1] + 2;
                else
                    dp[i][j] = max(dp[i+1][j], dp[i][j-1]);
            }
        }
        return dp[0][s.length()-1];
    }
};